src/util/RtreeBasicGeometry.h

```diff
@@ -0,0 +1,171 @@
+    auto minX = [](BasicGeometry::BoundingBox b) -> double {
+      return b.min_corner().get<0>();
+    };
+    auto minY = [](BasicGeometry::BoundingBox b) -> double {
+      return b.min_corner().get<1>();
+    };
+    auto maxX = [](BasicGeometry::BoundingBox b) -> double {
+      return b.max_corner().get<0>();
+    };
+    auto maxY = [](BasicGeometry::BoundingBox b) -> double {
+      return b.max_corner().get<1>();
+    };
+
+    double globalMinX = minX(b1) < minX(b2) ? minX(b1) : minX(b2);
+    double globalMinY = minY(b1) < minY(b2) ? minY(b1) : minY(b2);
+    double globalMaxX = maxX(b1) > maxX(b2) ? maxX(b1) : maxX(b2);
+    double globalMaxY = maxY(b1) > maxY(b2) ? maxY(b1) : maxY(b2);
+
+    return {{globalMinX, globalMinY}, {globalMaxX, globalMaxY}};
+  }
+
+  static bool IsBorderOfSplitCandidate(uint64_t current, uint64_t splitSize,
+                                       uint64_t M) {
+    if (((current + 1) % splitSize == 0 && (current + 1) / splitSize < M) ||
+        (current % splitSize == 0 && current / splitSize >= 1))
+      return true;
+    return false;
+  }
+
+  // ___________________________________________________________________________
+  // Convert a single wkt literal to a datapoint in the format suitable for the
+  // Rtree
+  static std::optional<BoundingBox> ConvertWordToRtreeEntry(
+      const std::string& wkt) {
+    /**
+     * Convert a single wkt literal to a boundingbox.
+     * Get the bounding box of either a multipolygon, polygon or a linestring
+     */
+    if (!wkt.starts_with("\"MULTIPOLYGON") && !wkt.starts_with("\"POLYGON") &&
+        !wkt.starts_with("\"LINESTRING")) {
+      return {};
+    }
+
+    double maxDouble = std::numeric_limits<double>::max();
+
+    double minX = maxDouble;
+    double maxX = -maxDouble;
+    double minY = maxDouble;
+    double maxY = -maxDouble;
+
+    // Iterate over matches and capture x and y coordinates
+    for (
+        auto match : ctre::range<
+            R"( *([\-|\+]?[0-9]+(?:[.][0-9]+)?) +([\-|\+]?[0-9]+(?:[.][0-9]+)?))">(
+            wkt)) {
+      double x = std::stod(std::string(match.get<1>()));
+      double y = std::stod(std::string(match.get<2>()));
+
+      if (x < minX) minX = x;
+      if (x > maxX) maxX = x;
+      if (y < minY) minY = y;
+      if (y > maxY) maxY = y;
+    }
+
+    return {BasicGeometry::CreateBoundingBox(minX, minY, maxX, maxY)};
+  }
+};
+
+// ___________________________________________________________________________
+// Data type for a value of the Rtree, which contains the id of the object and
+// its bounding box.
+struct RTreeValue {
+  BasicGeometry::BoundingBox box{};
+  uint64_t id = 0;
+  [[nodiscard]] double MinX() const { return box.min_corner().get<0>(); }
+  [[nodiscard]] double MaxX() const { return box.max_corner().get<0>(); }
+  [[nodiscard]] double MinY() const { return box.min_corner().get<1>(); }
+  [[nodiscard]] double MaxY() const { return box.max_corner().get<1>(); }
```

Comment:
Again, you ave this functionality, reuse it (the first `MinX` function in this file.).